### PR TITLE
Refactor db.py

### PR DESCRIPTION
**Pull request by zxenonx**
Fixes https://linear.app/zurichat2/issue/ZUR-78/refactor-dbpy

### CODE REVIEW

File: backend/utils/db.py [hunk 1]
```diff
@@ -1,51 +1,97 @@
...
         try:
-            response = requests.get(url=f"{BASE_URL}/marketplace/plugins")
+            response = requests.get(url=f"{settings.BASE_URL}/marketplace/plugins")
             plugins = response.json().get("data").get("plugins")
             plugin = next(
-                item for item in plugins if PLUGIN_KEY in item["template_url"]
+                item for item in plugins if settings.PLUGIN_KEY in item["template_url"]
             )
-            self.plugin_id = plugin["id"] if plugin else None
+            self.plugin_id = plugin.get("id")
```
Comment (o4codes) [inline backend/utils/db.py:73]: NIceeee

File: backend/utils/db.py [hunk 2]
```diff
@@ -1,51 +1,97 @@
...
 
         try:
-            response = requests.get(url=f"{BASE_URL}/marketplace/plugins")
+            response = requests.get(url=f"{settings.BASE_URL}/marketplace/plugins")
             plugins = response.json().get("data").get("plugins")
             plugin = next(
-                item for item in plugins if PLUGIN_KEY in item["template_url"]
+                item for item in plugins if settings.PLUGIN_KEY in item["template_url"]
             )
-            self.plugin_id = plugin["id"] if plugin else None
```
Comment (o4codes) [inline backend/utils/db.py:30]: Nicee

**Review by o4codes — APPROVED** [review]
Good work @zxenonx 
This is really good.
Thanks for the refactor

**Review by mikenrowland — APPROVED** [review]
LGTM

File: backend/utils/db.py [hunk 3]
```diff
@@ -91,57 +165,107 @@ async def update(self, collection_name, document_id, data):
...
-            "plugin_id": self.plugin_id,
+            "filter": query,
             "options": options,
         }
 
         try:
-            response = requests.post(url=self.read_query_api, json=body)
+            response = requests.post(url=self.read_api, json=body)
         except requests.exceptions.RequestException as exception:
             print(exception)
```
Comment (paulinebanye) [inline backend/utils/db.py]: Print statement here

File: backend/utils/db.py [hunk 4]
```diff
@@ -158,56 +282,82 @@ async def delete(self, collection_name, document_id):
...
+        Raises:
+            RequestException: Unable to connect to zc_core
         """
-        url = self.get_member_api.format(org_id=self.organization_id)
+
+        url = self.get_members_api.format(org_id=self.organization_id)
         try:
             response = requests.get(url=url)
         except requests.exceptions.RequestException as exception:
             print(exception)
```
Comment (paulinebanye) [inline backend/utils/db.py]: Print statement here

File: backend/utils/file_storage.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+            self.upload_multiple_api = (
+                f"{settings.BASE_URL}/upload/files/" + "{self.plugin_id}"
+            )
+            self.delete_file_api = (
+                f"{settings.BASE_URL}/delete/file/" + "{self.plugin_id}"
+            )
+            self.organization_id = organization_id
+
+        except requests.exceptions.RequestException as exception:
+            print(exception)
```
Comment (paulinebanye) [inline backend/utils/file_storage.py:30]: Print statement here

**Review by man-of-mind — APPROVED** [review]

**Review by paulinebanye — CHANGES_REQUESTED** [review]

**Review by paulinebanye — COMMENTED** [review]
Well done X, as discussed I've outlined where the print statements are

File: backend/utils/db.py (hunk 3 above)
Comment (mikenrowland) [reply backend/utils/db.py]: what is wrong with the print statement? @pauline-banye

Comment (zxenonx) [reply backend/utils/db.py]: @mykie88, we should have got rid of all of them before deploying to production.
In fact, the print() statement is only meant for debugging in a local env (the use of the debugger + breakpoints is more efficient though).

It becomes difficult to remove/disable all the print() statements when we are done with debugging and that's why there's a dedicated tool for logging. 
This is a built-in module called **logging** that provides tracking for events that occur while the app runs (different levels of severity can be set, debug-warning-info-critical, etc.).

For now, I'm just going to use the raise-from syntax.

Comment (pauline-banye) [reply backend/utils/db.py:228]: It was brought up in a different PR and it had to be removed. Have the rules changed?

**Review by paulinebanye — APPROVED** [review]
Seen the changes, good job X

File: backend/utils/db.py [hunk 6]
```diff
@@ -1,51 +1,97 @@
...
+                item for item in plugins if settings.PLUGIN_KEY in item["template_url"]
             )
-            self.plugin_id = plugin["id"] if plugin else None
+            self.plugin_id = plugin.get("id")
         except requests.exceptions.RequestException as exception:
-            print(exception)
             raise HTTPException(
                 status_code=status.HTTP_408_REQUEST_TIMEOUT, detail="Request Timeout"
-            )
+            ) from exception
```
Comment (zxenonx) [inline backend/utils/db.py]: @man-of-mind, we need to change this part.
When a Request exception occurs, the reason could be, not only a timeout but also an HTTP error, a Connection error, etc. 
Do we want to handle each type of RequestException explicitly?

Comment (man-of-mind) [reply backend/utils/db.py]: If there are possibilities of other exceptions, it is better we handle them differently.
@zxenonx

Comment (zxenonx) [reply backend/utils/db.py]: https://docs.python-requests.org/en/master/_modules/requests/exceptions/

@man-of-mind

Comment (man-of-mind) [reply backend/utils/db.py]: Okay boss, should i do this or u will add it to this PR?
@zxenonx

Comment (zxenonx) [reply backend/utils/db.py]: I'm gonna do it. @man-of-mind

Comment (man-of-mind) [reply backend/utils/db.py]: Alright boss @zxenonx

Comment (zxenonx) [reply backend/utils/db.py]: I've made the changes. Please check.

**Review by man-of-mind — COMMENTED** [review]
Well done boss @zxenonx

**Pull request merged**